--- tcl8.4.6/unix/configure	2004-03-01 23:58:59.000000000 +0000
+++ tcl8.4.6.new/unix/configure	2007-08-02 21:53:03.000000000 +0100
@@ -1,4 +1,4 @@
-#! /bin/sh
+#! /bin/bash
 
 # Guess values for system-dependent variables and create Makefiles.
 # Generated automatically using autoconf version 2.13 
@@ -5538,7 +5538,8 @@
     # for the system.  This can usually be done via the "uname" command, but
     # there are a few systems, like Next, where this doesn't work.
 
-    echo $ac_n "checking system version (for dynamic loading)""... $ac_c" 1>&6
+
+    echo $ac_n "checking system version (for dynamic loading)" "... $ac_c" 1>&6
 echo "configure:5543: checking system version (for dynamic loading)" >&5
     if test -f /usr/lib/NextStep/software_version; then
 	system=NEXTSTEP-`awk '/3/,/3/' /usr/lib/NextStep/software_version`
