--- squidGuard-1.5-beta/src/sg.h-ORIG	2013-09-27 01:03:43.000000000 -0400
+++ squidGuard-1.5-beta/src/sg.h	2013-09-29 20:57:18.000000000 -0400
@@ -2,7 +2,7 @@
   By accepting this notice, you agree to be bound by the following
   agreements:
   
-  This software product, squidGuard, is copyrighted (C) 1998-2009
+  This software product, squidGuard, is copyrighted (C) 1998-2007
   by Christine Kronberg, Shalla Secure Services. All rights reserved.
  
   This program is free software; you can redistribute it and/or modify it
@@ -68,13 +68,12 @@
 #define ACL_TYPE_DEFAULT    1
 #define ACL_TYPE_TERMINATOR 2
 #define ACL_TYPE_INADDR     3
-#define ACL_TYPE_DNSBL      4
 
 #define REQUEST_TYPE_REWRITE    1
 #define REQUEST_TYPE_REDIRECT   2
 #define REQUEST_TYPE_PASS       3
 
-#define MAX_BUF 12288
+#define MAX_BUF 4096
 
 #define DEFAULT_LOGFILE "squidGuard.log"
 #define WARNING_LOGFILE "squidGuard.log"
@@ -87,7 +86,7 @@
 
 #ifdef ACCONFIG
 #undef DEFAULT_CONFIGFILE
-#define DEFAULT_CONFIGFILE "/usr/local/squidGuard/squidGuard.conf"
+#define DEFAULT_CONFIGFILE "/var/smoothwall/urlfilter/squidGuard.conf"
 #endif
 
 #ifdef ACLOGDIR
@@ -223,29 +222,6 @@
   struct Ip *next;
 };
 
-/* ldapip */
-#ifdef HAVE_LIBLDAP
-  /* LDAP tracking */
-struct IpInfo {
-  /* quota tracking */
-  time_t time;
-  time_t last;
-  int consumed;
-  char status;
-  int ldapip;                        /* bool: 1 if ip loaded from LDAP */
-  int found;                   /* bool: we also cache if not found in LDAP */
-  time_t cachetime;            /* time this item was added to cache */
-};
-#endif
-
-struct IpQuota {
-  time_t seconds;
-  int renew;
-  time_t sporadic;
-};
-/* ldapip */
-
-
 struct Setting {
   char *name;
   char *value;
@@ -272,7 +248,6 @@
 };
 
 struct Destination {
-  //int syslogStatus;
   char *name;
   int active;
   char *domainlist;
@@ -302,12 +277,8 @@
   struct UserQuota userquota;
   struct LogFile *logfile;
 #ifdef HAVE_LIBLDAP
-  struct sgDb *ipDb;
-  struct IpQuota ipquota;  
-  char **ldapuserurls;                     /* dynamic array of url strings */
-  int ldapuserurlcount;                    /* current size of pointer array */
-  char **ldapipurls;                     /* dynamic array of url strings */
-  int ldapipurlcount;                    /* current size of pointer array */
+  char **ldapurls;                     /* dynamic array of url strings */
+  int ldapurlcount;                    /* current size of pointer array */
 #endif
   struct Source *next;
 };
@@ -330,7 +301,6 @@
 
 struct AclDest {
   char *name;
-  char *dns_suffix;
   struct Destination *dest;
   int    access;
   int    type;
@@ -347,11 +317,8 @@
 
 void   sgReadConfig __P((char *));
 void   sgLog __P((struct LogFileStat *, char *, ...));
-void   sgLogDebug __P((char *, ...));
-void   sgLogNotice __P((char *, ...));
-void   sgLogWarn __P((char *, ...));
 void   sgLogError __P((char *, ...));
-void   sgLogFatal __P((char *, ...));
+void   sgLogFatalError __P((char *, ...));
 void   sgSetGlobalErrorLogFile __P(());
 void   sgLogRequest __P((struct LogFile *, struct SquidInfo *, struct Acl *, struct AclDest *, struct sgRewrite *, int));
 int    parseLine __P((char *, struct SquidInfo *));
@@ -373,7 +340,6 @@
 void   sgSourceUserList __P((char *));
 #ifdef HAVE_LIBLDAP
 void   sgSourceLdapUserSearch __P((char *));
-void   sgSourceLdapIpSearch __P((char *)); /* ldapip */
 #endif
 void   sgSourceExecUserList __P((char *));
 void   sgSourceDomain __P((char *));
@@ -393,9 +359,6 @@
 void   sgDestTime __P((char *, int));
 
 void   sgSetting __P((char *, char *));
-#ifdef USE_SYSLOG
-void   sgSyslogSetting __P((char *));
-#endif
 struct Setting *sgSettingFindName __P((char *));
 char   *sgSettingGetValue __P((char *));
 
@@ -449,8 +412,6 @@
 struct UserQuotaInfo *setuserquota __P(());
 void sgSourceUserQuota __P((char *, char *, char *));
 
-struct IpQuotaInfo *setipquota __P(()); /* ldapip */
-void sgSourceIpQuota __P((char *, char *, char *)); /*ldapip */
 
 void   *sgMalloc __P((size_t));
 void   *sgCalloc __P((size_t, size_t));
@@ -478,7 +439,6 @@
 int    sgFindUser __P((struct Source *, char *, struct UserInfo **));
 #ifdef HAVE_LIBLDAP
 int    sgDoLdapSearch __P((const char *, const char *));
-int    sgFindIp __P((struct Source *, char *, struct IpInfo **));
 #endif
 
 int    expand_url __P((char *, size_t, const char *, const char *));
