diff -Nurb eagle-usb-2.3.2/driver/eu_main.c eagle-usb-2.3.2.new/driver/eu_main.c
--- eagle-usb-2.3.2/driver/eu_main.c	2005-05-20 14:27:45.000000000 +0100
+++ eagle-usb-2.3.2.new/driver/eu_main.c	2005-09-05 14:58:30.000000000 +0100
@@ -363,10 +363,11 @@
     
     eu_enters (DBG_INIT);
     
-    eu_dbg (DBG_INIT,"vid (%#X) pid (%#X) \n",
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,11)
+    eu_dbg (DBG_INIT,"vid (%#X) pid (%#X) \n",
              le16_to_cpu(usb->descriptor.idVendor), le16_to_cpu(usb->descriptor.idProduct));
 #else
+    eu_dbg (DBG_INIT,"vid (%#X) pid (%#X) \n",
              usb->descriptor.idVendor, usb->descriptor.idProduct);
 #endif
 
