diff -Nurb openswan-1.0.6.old/utils/ipsec.in openswan-1.0.6/utils/ipsec.in
--- openswan-1.0.6.old/utils/ipsec.in	2003-11-25 00:37:27.000000000 +0000
+++ openswan-1.0.6/utils/ipsec.in	2004-10-29 20:03:20.000000000 +0100
@@ -20,7 +20,7 @@
 export IPSEC_DIR IPSEC_CONFS
 
 # standardize PATH, and export it for everything else's benefit
-PATH=/usr/local/sbin:/sbin:/usr/sbin:/usr/local/bin:/bin:/usr/bin
+PATH=/usr/local/sbin:/sbin:/usr/sbin:/usr/local/bin:/bin:/usr/bin:/modules/tunnel/usr/sbin
 export PATH
 
 # things not to be listed in --help command list
diff -Nurb openswan-1.0.6.old/utils/setup openswan-1.0.6/utils/setup
--- openswan-1.0.6.old/utils/setup	2003-02-21 23:35:07.000000000 +0000
+++ openswan-1.0.6/utils/setup	2004-10-29 20:05:35.000000000 +0100
@@ -37,7 +37,7 @@
 if test " $IPSEC_DIR" = " "	# if we were not called by the ipsec command
 then
 	# we must establish a suitable PATH ourselves
-	PATH=/usr/local/sbin:/sbin:/usr/sbin:/usr/local/bin:/bin:/usr/bin
+	PATH=/usr/local/sbin:/sbin:/usr/sbin:/usr/local/bin:/bin:/usr/bin:/modules/tunnel/usr/sbin
 	export PATH
 fi
 
