diff -Nurb openswan-1.0.6.old/utils/_realsetup openswan-1.0.6/utils/_realsetup
--- openswan-1.0.6.old/utils/_realsetup	2004-01-11 16:34:55.000000000 +0000
+++ openswan-1.0.6/utils/_realsetup	2004-10-29 20:08:08.000000000 +0100
@@ -70,6 +70,12 @@
 	i=`lsmod 2>&1 | awk '$1 == "ipsec" { print $1 }'`
 	if test " $i" = " ipsec"
 	then
+                rmmod ipsec_aes >/dev/null 2>/dev/null
+                rmmod ipsec_cast >/dev/null 2>/dev/null
+                rmmod ipsec_sha1 >/dev/null 2>/dev/null
+                rmmod ipsec_sha2 >/dev/null 2>/dev/null
+                rmmod ipsec_twofish >/dev/null 2>/dev/null
+                rmmod ipsec_blowfish >/dev/null 2>/dev/null
 		rmmod ipsec
 	fi
 }
diff -Nurb openswan-1.0.6.old/utils/_startklips openswan-1.0.6/utils/_startklips
--- openswan-1.0.6.old/utils/_startklips	2002-11-05 01:35:58.000000000 +0000
+++ openswan-1.0.6/utils/_startklips	2004-10-29 20:08:15.000000000 +0100
@@ -221,6 +221,12 @@
 then
     # statically compiled KLIPS not found; try to load the module
     insmod ipsec
+    insmod ipsec_aes
+    insmod ipsec_cast
+    insmod ipsec_sha1
+    insmod ipsec_sha2
+    insmod ipsec_twofish
+    insmod ipsec_blowfish
 fi
 
 if test ! -f $ipsecversion
@@ -231,6 +237,12 @@
 		unset MODPATH MODULECONF	# no user overrides!
 		depmod -a >/dev/null 2>&1
 		modprobe -v ipsec
+		modprobe ipsec_aes
+		modprobe ipsec_cast
+		modprobe ipsec_sha1
+		modprobe ipsec_sha2
+		modprobe ipsec_twofish
+		modprobe ipsec_blowfish
 	fi
 	if test ! -f $ipsecversion
 	then
