--- linux-2.6.16.60.old/drivers/scsi/ahci.c	2010-02-01 15:37:48.000000000 +0000
+++ linux-2.6.16.60/drivers/scsi/ahci.c	2010-02-01 15:43:01.000000000 +0000
@@ -367,6 +367,22 @@
 	  board_ahci }, /* ICH10 */
 	{ PCI_VENDOR_ID_INTEL, 0x3a22, PCI_ANY_ID, PCI_ANY_ID, 0, 0,
 	  board_ahci }, /* ICH10 Gigabyte EG45M-DS2H */
+	{ PCI_VENDOR_ID_INTEL, 0x3b22, PCI_ANY_ID, PCI_ANY_ID, 0, 0,
+	  board_ahci }, /* PCH AHCI */
+	{ PCI_VENDOR_ID_INTEL, 0x3b23, PCI_ANY_ID, PCI_ANY_ID, 0, 0,
+	  board_ahci }, /* PCH AHCI */
+	{ PCI_VENDOR_ID_INTEL, 0x3b24, PCI_ANY_ID, PCI_ANY_ID, 0, 0,
+	  board_ahci }, /* PCH RAID */
+	{ PCI_VENDOR_ID_INTEL, 0x3b25, PCI_ANY_ID, PCI_ANY_ID, 0, 0,
+	  board_ahci }, /* PCH RAID */
+	{ PCI_VENDOR_ID_INTEL, 0x3b29, PCI_ANY_ID, PCI_ANY_ID, 0, 0,
+	  board_ahci }, /* PCH AHCI */
+	{ PCI_VENDOR_ID_INTEL, 0x3b2b, PCI_ANY_ID, PCI_ANY_ID, 0, 0,
+	  board_ahci }, /* PCH RAID */
+	{ PCI_VENDOR_ID_INTEL, 0x3b2c, PCI_ANY_ID, PCI_ANY_ID, 0, 0,
+	  board_ahci }, /* PCH RAID */
+	{ PCI_VENDOR_ID_INTEL, 0x3b2f, PCI_ANY_ID, PCI_ANY_ID, 0, 0,
+	  board_ahci }, /* PCH AHCI */
 	{ 0x197b, 0x2360, PCI_ANY_ID, PCI_ANY_ID, 0, 0,
 	  board_ahci }, /* JMicron JMB360 */
 	{ 0x197b, 0x2361, PCI_ANY_ID, PCI_ANY_ID, 0, 0,
