--- linux-2.6.23-git12/include/asm-i386/mach-default/mach_apicdef.h	2009-07-27 13:09:22.000000000 +0100
+++ linux-2.6.23-git13/include/asm-i386/mach-default/mach_apicdef.h	2009-07-27 13:09:33.000000000 +0100
@@ -1,10 +1,16 @@
 #ifndef __ASM_MACH_APICDEF_H
 #define __ASM_MACH_APICDEF_H
 
+#include <asm/apic.h>
+
 #define		APIC_ID_MASK		(0xF<<24)
 
 static inline unsigned get_apic_id(unsigned long x) 
 { 
+	unsigned int ver = GET_APIC_VERSION(apic_read(APIC_LVR));
+	if (APIC_XAPIC(ver))
+		return (((x)>>24)&0xFF);
+	else
 	return (((x)>>24)&0xF);
 } 
 
