--- linux-2.6.16.62/drivers/message/fusion/mptscsih.c	2008-07-21 19:48:10.000000000 +0000
+++ linux-2.6.16.60/drivers/message/fusion/mptscsih.c	2009-07-15 13:14:16.000000000 +0000
@@ -744,10 +744,7 @@
 			sc->resid=0;
 		case MPI_IOCSTATUS_SCSI_RECOVERED_ERROR:	/* 0x0040 */
 		case MPI_IOCSTATUS_SUCCESS:			/* 0x0000 */
-			if (scsi_status == MPI_SCSI_STATUS_BUSY)
-				sc->result = (DID_BUS_BUSY << 16) | scsi_status;
-			else
-				sc->result = (DID_OK << 16) | scsi_status;
+			sc->result = (DID_OK << 16) | scsi_status;
 			if (scsi_state == 0) {
 				;
 			} else if (scsi_state & MPI_SCSI_STATE_AUTOSENSE_VALID) {
